### PR TITLE
cli tool: reimplement stdin buffering in -F.

### PR DESCRIPTION
**Pull request by monnerat**
Dummy PR to check new code on APPVeyor/Travis CI

### CODE REVIEW

File: src/tool_formparse.c [hunk 1]
```diff
@@ -372,16 +383,146 @@ static int get_param_part(struct OperationConfig *config, char **str,
   return sep & 0xFF;
 }
 
-/* Check if file is "-". If so, use a callback to read OUR stdin (to
+
+/* Mime part callbacks for stdin. */
+static size_t stdin_read(char *buffer, size_t size, size_t nitems, void *arg)
```
Comment (bagder) [inline src/tool_formparse.c]: There's already a [file2memory](https://github.com/curl/curl/blob/master/src/tool_paramhlp.c#L90) function you should use instead!

File: src/tool_formparse.c [hunk 2]
```diff
@@ -372,16 +383,146 @@ static int get_param_part(struct OperationConfig *config, char **str,
...
+      if(!p) {
+        stdin_free(sip);
+        return CURLE_OUT_OF_MEMORY;
+      }
+      sip->data = p;
+      sip->curpos = sip->size + STDIN_BUFSIZE;
+    }
+    /* Shrink buffer to spare memory resources. */
+    if(sip->size < sip->curpos)
+      sip->data = realloc(sip->data, (size_t) sip->size);
```
Comment (bagder) [inline src/tool_formparse.c]: Even shrinking can fail so this return code must be checked and bail out if you fail.

**Comment by monnerat** [comment]
Pushed: https://github.com/curl/curl/commit/665b3e48bc95713ecceaddf445fc2049440a6fb3

**Comment by bagder** [comment]
Yes, but none of my remarks were addressed?

And please use 'closes #[num]' in commit messages that close PRs (or "closes [URL]")

**Comment by bagder** [comment]
oh, did I remark _after_ it was merged perhaps?

**Comment by monnerat** [comment]
> oh, did I remark after it was merged perhaps? I think you need to give us slightly more time to review...
Yes, I think so: I've just received the mails.

Thanks for your review and I'll take your remarks in consideration. I was not expecting it but just wanted to have checks on cross-platforms :-)

**Comment by monnerat** [comment]
> There's already a file2memory function you should use instead!

Fixed: https://github.com/curl/curl/commit/56509055d23986479f472d667625426f51074992

**Pull request closed without merging**